### DISCUSSION TITLE
Hosting mitmproxy as a web service?

### DISCUSSION THREAD

**Discussion by jaimeiniesta** [mitmproxy/mitmproxy#7312] [Q&A]
Hi, how can I run mitmproxy as a web service?

I've tried deploying the [Docker image](https://hub.docker.com/r/mitmproxy/mitmproxy/) to Render.com but the installation fails saying that a interactive shell environment is required.

Is it possible to run mitmproxy in a non-interactive shell?

```
==> Starting service...
usermod: no changes
Error: mitmproxy's console interface requires a tty. Please run mitmproxy in an interactive shell environment.
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/mitmproxy/master.py", line 80, in run
    await self.running()
  File "/usr/local/lib/python3.11/site-packages/mitmproxy/tools/console/master.py", line 207, in running
    sys.exit(1)
SystemExit: 1
```

**Comment by Prinzhorn** [comment]
> mitmdump is the command-line version of mitmproxy. Think tcpdump for HTTP.

https://docs.mitmproxy.org/stable/#3-powerful-core-tools
